### PR TITLE
SSCS-10218 Staff reference interface added

### PR DESCRIPTION
**Pull request by Valantinas**
Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10218

### Change description ###
Added feign client and related model classes for the caseworker reference data

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/client/CaseWorkerRefDataApi.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import uk.gov.hmcts.reform.sscs.model.ServiceRoleMapping;
+import uk.gov.hmcts.reform.sscs.model.UserRequest;
+
+
+@FeignClient(
+    name = "caseworker-ref-data-api",
+    url = "${caseworker-ref.api.url}"
+)
+public interface CaseWorkerRefDataApi {
+    String SERVICE_AUTHORIZATION = "serviceAuthorization";
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/client/CaseWorkerRefDataApi.java:26]: Should this have an access modifier? (Same as below)

File: src/main/java/uk/gov/hmcts/reform/sscs/model/ServiceRoleMapping.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class ServiceRoleMapping extends CaseWorkerDomain {
+
+    @MappingField(columnName = "Service ID", position = 1)
+    @NotEmpty
+    @JsonProperty(value = "service_code")
+    @JsonDeserialize(using = TrimStringFields.class)
+    String serviceId;
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/ServiceRoleMapping.java:26]: Same access modifier question here and below

File: src/main/java/uk/gov/hmcts/reform/sscs/utility/TrimStringFields.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+public class TrimStringFields extends JsonDeserializer<Object> {
+    @Override
+    public Object deserialize(JsonParser jp, DeserializationContext dc) {
+        JsonNode node;
+        try {
+            node = jp.readValueAsTree();
+            if (nonNull(node) && node.isTextual() && isNotEmpty(node.asText())) {
+                return node.asText().trim();
+            }
+        } catch (IOException e) {
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/utility/TrimStringFields.java:24]: We should add a negative test case

File: src/test/java/uk/gov/hmcts/reform/sscs/utility/TrimStringFieldsTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import java.io.IOException;
+import org.junit.jupiter.api.Test;
+
+
+class TrimStringFieldsTest {
+
+    TrimStringFields underTest = new TrimStringFields();
```
Comment (dattipoeHMCTS) [inline src/test/java/uk/gov/hmcts/reform/sscs/utility/TrimStringFieldsTest.java:17]: Access modifiers

File: src/main/java/uk/gov/hmcts/reform/sscs/model/ServiceRoleMapping.java (hunk 2 above)
Comment (Valantinas) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/ServiceRoleMapping.java:26]: this is a copy/paste class from the reference API - one day these models should be shared between projects, so for now I decided not to touch it:
https://github.com/hmcts/rd-caseworker-ref-api/blob/master/src/main/java/uk/gov/hmcts/reform/cwrdapi/client/domain/ServiceRoleMapping.java#L27

File: src/test/java/uk/gov/hmcts/reform/sscs/utility/TrimStringFieldsTest.java (hunk 4 above)
Comment (Valantinas) [reply src/test/java/uk/gov/hmcts/reform/sscs/utility/TrimStringFieldsTest.java:17]: access modifiers are not needed for test classes - they are not deployed to production

File: src/main/java/uk/gov/hmcts/reform/sscs/client/CaseWorkerRefDataApi.java (hunk 1 above)
Comment (Valantinas) [reply src/main/java/uk/gov/hmcts/reform/sscs/client/CaseWorkerRefDataApi.java:26]: access modifiers are not needed for interface methods or variables. 
this has also been done in the existing interface `RefDataApi.java`

File: src/main/java/uk/gov/hmcts/reform/sscs/utility/TrimStringFields.java (hunk 3 above)
Comment (Valantinas) [reply src/main/java/uk/gov/hmcts/reform/sscs/utility/TrimStringFields.java:24]: negative case added (the project I copied this class from didn't have any tests at all)

File: src/main/java/uk/gov/hmcts/reform/sscs/client/CaseWorkerRefDataApi.java (hunk 1 above)
Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/client/CaseWorkerRefDataApi.java:26]: Ahaha, wow. I completely missed it was an interface

Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/client/CaseWorkerRefDataApi.java:26]: No as it's an interface - I suggest we should refactor these clients at a later stage to put this string in a constants class as we all have it in the interfaces.

**Review by dattipoeHMCTS — APPROVED** [review]

**Pull request merged**